### PR TITLE
wip(web): GraphQL

### PR DESCRIPTION
**Pull request by robzienert**
⚠️ This isn't intended for anything more than an exhibit.

We're seeing performance issues elsewhere and suspect we've gotten to the point where we're just transferring too much data out of front50 too often. GraphQL seems a logical choice over something like more endpoints or query params to change response payloads.

I'd say I'm... 10% sure what I'm doing at this point, but is this intriguing enough to pursue?

screen shot 2018-10-02 at 6 14 45 pm

@spinnaker/netflix-reviewers

### CODE REVIEW

File: front50-web/src/main/resources/graphql/pipelineConfig.graphqls [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  triggers: [Trigger]!
+}
+
+type Artifact {
+  type: String!
+  name: String!
+  version: String!
+  location: String!
+  reference: String!
+  # cannot model metadata
```
[the remark this answers is not in the document]
Comment (robzienert) [reply front50-web/src/main/resources/graphql/pipelineConfig.graphqls]: Got around this with a JSON scalar type. Kinda weird, but it works.

**Comment by robzienert** [comment]
Pretty sure I managed to model a full pipeline as it exists today? Thoughts:

1. Would want to pull out into a new `front50-graphql` module.
2. GraphQL configuration is a pain, might want to use something like `front50-[graphql_or_api]-client` that other services can pull in. Maybe https://github.com/apollographql/apollo-android could help?
3. If we're defining a schema like this, do we want to consider going deeper with gRPC and https://github.com/google/rejoiner? This would also solve our client problem.
4. The version of `graphql-java` I'm using is quite old and seems to be missing some nice-ities: It's the newest we can use on Spring Boot 1.x

**Comment by dreynaud** [comment]
Can't believe you got this done in one day! Super exciting

**Comment by robzienert** [comment]
Well, it's not done-done. I gotta like, make it into something that's mergeable and has tests, etc. :)

**Pull request merged**